### ISSUE TITLE
Add parameter to allow docker layer caching

### ISSUE THREAD

**Issue by dkarp0** [CircleCI-Public/aws-ecr-orb#67]
## What would you like to be added
Add parameter to allow docker layer caching
```
  - when:
      condition: <<parameters.setup-remote-docker>>
      steps:
        - setup_remote_docker:
           docker_layer_caching: <<parameters.docker-layer-caching>>
```
## Why is this needed

When using a custom docker executor, setup_remote_docker is most useful when you can enable docker_layer_caching to speed up builds. This parameter is `false` by default, so we should have a way to enable it in the orb parameters. Otherwise it's impossible to enable when using this orb.

**Comment by dankarpinski** [comment]
found a way to do this using the machine executor:
```
executors:
  build-executor:
    machine:
      docker_layer_caching: true
```

**Comment by adamrp** [comment]
But how do you use DLC with a _custom docker executor_ as your original question stated? e.g., if my config.yml file looks something like this:
```
version: 2.1
orbs:
  aws-ecr: circleci/aws-ecr@6.1.0
  aws-ecs: circleci/aws-ecs@0.0.8
executors:
  ubuntu-lfs:
    docker:
      - image: "${AWS_ECR_ACCOUNT_URL}/circle-executor:latest"
workflows:
  build-and-deploy:
    jobs:
      - aws-ecr/build-and-push-image:
          executor: ubuntu-lfs
          setup-remote-docker: true
... SNIP ...                       
```
The only argument that `setup-remote-docker` accepts is a Boolean (`true` in this case), correct? and the `docker` executor type doesn't have a parameter for enabling DLC?

**Comment by dankarpinski** [comment]
Luckily wasn't necessary for my needs. It isn't possible now with a custom docker executor.

**Comment by iynere** [comment]
@dankarpinski is there a reason you closed your PR back in October? we just merged a similar change to our Docker orb:

https://github.com/CircleCI-Public/docker-orb/pull/44

feel free to re-open or open a new PR and we will get it merged ASAP—thanks !

**Comment by PavlosMelissinos** [comment]
@dankarpinski I have the same question as @iynere 

I'd appreciate it if the PR could be merged!

**Comment by dankarpinski** [comment]
Re-opened as a few people have mentioned demand. @iynere could you merge?

**Comment by a7i** [comment]
@dankarpinski I created a new PR which also conforms with the naming convention used for remote docker version (i.e. `remote-docker-version`). PR #116

**Comment by KyleTryon** [comment]
Resolved in #116. Thank you.
